Sources/swift-algorand-sdk/v2/client/Request.swift

```diff
@@ -0,0 +1,119 @@
+        self.queryParameters = queryParameters
+        self.bodyParameters = bodyParameters
+        self.encoding = encoding
+    }
+    
+    func request(from client: HTTPClient) -> DataRequest {
+        var components = client.connectString()
+        components.path += self.path
+        components.queryItems = queryParameters.map { URLQueryItem(name: $0.key, value: $0.value) }
+        let headers = self.headers.merging(client.defaultHTTPHeaders) { _, global in global  }
+        return client.session.request(components.url?.absoluteString ?? "",
+                          method: method,
+                          parameters: bodyParameters,
+                          headers: .init(headers),
+                          requestModifier: { $0.timeoutInterval = 120.0 })
+            .validate()
+    }
+}
+
+struct ByteEncoding: ParameterEncoding {
+  private let data: Data
+
+  init(data: Data) {
+    self.data = data
+  }
+
+  func encode(_ urlRequest: URLRequestConvertible, with parameters: Parameters?) throws -> URLRequest {
+    var urlRequest = try urlRequest.asURLRequest()
+    urlRequest.httpBody = data
+    return urlRequest
+  }
+}
+
+struct ResponseError: Codable {
+    let message: String?
+}
+
+public extension Request {
+
+    func execute( callback: @escaping (_:Response<ResponseType>) -> Void) {
+        let request = parameters.request(from: client)
+        let customResponse: Response<ResponseType> = Response()
+        request.responseDecodable(of: ResponseType.self) {  (response) in
+            if let error = response.error {
+                customResponse.setIsSuccessful(value: false)
+                let errorDescription = String(data: response.data ??
+                                              error.errorDescription?.data(using: .utf8) ??
+                                              Data(), encoding: .utf8)
+                customResponse.setErrorDescription(errorDescription: errorDescription ?? "")
+                
+                let decoder = JSONDecoder()
+                if let data = response.data,
+                   let errorObject = try? decoder.decode(ResponseError.self, from: data) {
+                    customResponse.errorMessage = errorObject.message
+                }
+                callback(customResponse)
+//                if let data = response.data,
+//                   let message = String(data: data, encoding: .utf8),
+//                   let dictionary = try? JSONSerialization.jsonObject(with: message.data, options: []) as? [String: Any] {
+//                    customResponse.errorMessage = dictionary["message"] as? String
+//                }
+                return
+            }
+            
+            guard let data = response.value else {
+                customResponse.errorMessage = "No valid data found"
+                customResponse.setIsSuccessful(value: true)
+                callback(customResponse)
+                return
+            }
```

Comment:
Just thinking, am not sure we want `customResponse.setIsSuccessful` to be set to true in this case, what do you think?

Reply:
agree, it's a classic copypaste mistake :)  Changing it right now
In previous versions, you had a `data!` force unwrap so i guess this is not a "real case" anyway